bleak/exc.py

```diff
@@ -9,6 +10,27 @@ class BleakError(Exception):
     pass
 
 
+class BluetoothPermissionErrorReason(enum.Enum):
+    # The bluetooth access was denied explicitly by the user.
+    Denied = enum.auto()
+
+    # The bluetooth access is restricted, possibly due to active restrictions such as parental controls.
+    Restricted = enum.auto()
+
+    # The bluetooth access was denied for an unknown reason
+    Unknown = enum.auto()
+
+
+class BleakBluetoothPermissionError(BleakError):
+    """
+    Exception which is raised if the Bluetooth access is denied.
+    """
+
+    def __init__(self, msg: str, reason: BluetoothPermissionErrorReason) -> None:
+        super().__init__(msg)
+        self.reason = reason
```

Comment:
```suggestion
        super().__init__(msg, reason)

        @property
        def reason(self) -> BluetoothPermissionErrorReason:
            return self.args[1]
```

I don't know why, but I like to handle custom exceptions like this. 

The property makes the `reason` attribute read-only and passing all args to the super class makes it print the reason enum as well as the message in the stack trace.